source/fuzz/transformation_add_parameter.cpp

```diff
@@ -81,17 +136,22 @@ void TransformationAddParameter::Apply(
 
   fuzzerutil::UpdateModuleIdBound(ir_context, message_.parameter_fresh_id());
 
-  // TODO(https://github.com/KhronosGroup/SPIRV-Tools/issues/3403):
-  //  Add an PointeeValueIsIrrelevant fact if the parameter is a pointer.
-
-  // Mark new parameter as irrelevant so that we can replace its use with some
-  // other id.
-  transformation_context->GetFactManager()->AddFactIdIsIrrelevant(
-      message_.parameter_fresh_id());
+  // If the |new_parameter_type_id| is not a pointer type, mark new parameter as
+  // irrelevant so that we can replace its use with some other id. If the
+  // |new_parameter_type_id| is not a pointer type, we cannot mark it, because
+  // this pointer might be replaced by a pointer from original shader. This
+  // would change the semantics of the module.
+  auto new_parameter_type =
+      ir_context->get_type_mgr()->GetType(new_parameter_type_id);
+  if (new_parameter_type->kind() != opt::analysis::Type::kPointer) {
+    transformation_context->GetFactManager()->AddFactIdIsIrrelevant(
+        message_.parameter_fresh_id());
+  }
```

Comment:
If we are adding a pointer parameter, we should mark it as PointeeValueIsIrrelevant.